packages/backend/src/models/ProjectModel.ts

```diff
@@ -54,48 +57,148 @@ export class ProjectModel {
         }));
     }
 
-    async updateCredentials(
-        projectUuid: string,
-        credentials: CreateWarehouseCredentials,
+    private async upsertWarehouseConnection(
+        trx: Transaction,
+        projectId: number,
+        data: CreateWarehouseCredentials,
     ): Promise<void> {
-        const projects = await this.database('projects')
-            .where('project_uuid', projectUuid)
-            .select('*');
-        if (projects.length === 0) {
-            throw new NotExistsError(
-                `No project exists with id ${projectUuid}`,
-            );
-        }
-        const [project] = projects;
         let encryptedCredentials: Buffer;
         try {
             encryptedCredentials = this.encryptionService.encrypt(
-                JSON.stringify(credentials),
+                JSON.stringify(data),
             );
         } catch (e) {
             throw new UnexpectedServerError('Could not save credentials.');
         }
-        await this.database('warehouse_credentials')
+        await trx('warehouse_credentials')
             .insert({
-                project_id: project.project_id,
-                warehouse_type: credentials.type,
+                project_id: projectId,
+                warehouse_type: data.type,
                 encrypted_credentials: encryptedCredentials,
             })
             .onConflict('project_id')
             .merge();
     }
 
+    async hasProjects(): Promise<boolean> {
```

Comment:
I guess this'll eventually be scoped to organisations?

Reply:
Once we support multiple orgs the health endpoint and the onboarding will have to change quite a bit.